homeassistant/components/dsmr/sensor.py

```diff
@@ -54,7 +54,7 @@
         vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.string,
         vol.Optional(CONF_HOST): cv.string,
         vol.Optional(CONF_DSMR_VERSION, default=DEFAULT_DSMR_VERSION): vol.All(
-            cv.string, vol.In(["5B", "5", "4", "2.2"])
+            cv.string, vol.In(["5L", "5B", "5", "4", "2.2"])
```

Comment:
Does this change yaml and violate ADR007? I'm working on a config flow and this small change fixes something that broke here #39664 during process of conversion, so I hope this is allowed anyway.

Reply:
The platform config is legacy. Affected users should set up the integration via config flow. Since the breakage has already been released there's no point in fixing it here.

Reply:
There is no config flow yet, I'm pushing changes incrementally (I have the next PR already open for some time). That means users can't fix their setup withour yaml.

I could add a config flow here, have the code ready but didn't to that to keep this small as it's more a bug fix which blocks people from updating.

Reply:
To be exact, the thing I try to fix here broke while making the import to ConfigEntry with a connection validation

Reply:
We've never allowed 5L here. How did that version work before?

Reply:
I'd rather add the user config flow than extending the schema. Please link the config flow PR and I'll have a look.

Reply:
> We've never allowed 5L here. How did that version work before?

Smarty follows it's own spec which happened to work because they mostly follow the Dutch spec. The one change is the serial id, it's reported differently. Since we validate connection we read the serial id to generate unique ids where before we didn't care about a serial. That's why and when it broke

Reply:
> I'd rather add the user config flow than extending the schema. Please link the config flow PR and I'll have a look.

There is no PR yet for config flow, I made a lot of changes which I'm pushing incrementally, so I was holding off with config flow until that merges.

I can add it here, the config flow code is not that much, it's only the code for the forms. It's probably more test code than actual code. Or do you rather have a separate PR?

Reply:
On second thought, let's keep this change and and be done for now.